sky/packages/sky/lib/src/material/material_list.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'constants.dart';
+import 'scrollbar_painter.dart';
+
+enum MaterialListType {
+  oneLine,
+  oneLineWithAvatar,
+  twoLine,
+  threeLine
+}
+
+Map<MaterialListType, double> _kItemExtent = const <MaterialListType, double>{
+  MaterialListType.oneLine: kOneLineListItemHeight,
+  MaterialListType.oneLineWithAvatar: kOneLineListItemWithAvatarHeight,
+  MaterialListType.twoLine: kTwoLineListItemHeight,
+  MaterialListType.threeLine: kThreeLineListItemHeight,
+};
+
+class MaterialList<T> extends StatefulComponent {
+  MaterialList({
+    Key key,
+    this.initialScrollOffset,
+    this.onScroll,
+    this.items,
+    this.itemBuilder,
+    this.type: MaterialListType.twoLine
+  }) : super(key: key);
+
+  final double initialScrollOffset;
+  final ScrollListener onScroll;
+  final List<T> items;
+  final ItemBuilder<T> itemBuilder;
+  final MaterialListType type;
+
+  _MaterialListState<T> createState() => new _MaterialListState<T>();
+}
+
+class _MaterialListState<T> extends State<MaterialList<T>> {
+
+  void initState() {
+    super.initState();
+    _scrollbarPainter = new ScrollbarPainter();
+  }
+
+  ScrollbarPainter _scrollbarPainter;
+
+  Widget build(BuildContext context) {
+    return new ScrollableList<T>(
+      initialScrollOffset: config.initialScrollOffset,
+      scrollDirection: ScrollDirection.vertical,
+      onScroll: config.onScroll,
+      items: config.items,
+      itemBuilder: config.itemBuilder,
+      itemExtent: _kItemExtent[config.type],
```

Comment:
> seems weird to be so specific about the type, but then not enforce it other than as the height.

Yeah, I played around with some variations but I didn't find anything that elegant.  One possibility might be to force a type specialization on `config.itemBuilder` (i.e., to `ListItem`) and then to wrap the builder in debug mode and assert that the built widget has a matching `type` property.  I didn't do that this patch because I haven't built out the other `ListItem` types yet.